### PR TITLE
std::distance bugfix in instruction::valid

### PR DESCRIPTION
**Pull request by CharlieL7**
* Fixes hanging issue when an argument to an instruction is from another module.
  * Assumes the argument is ordered correctly (before the current one) and does not check the instruction distance.
* Allows SSD-Resnet34 to compile
  * Model still stuck on allocation due to extremely high max_iterations for the loop instruction

### CODE REVIEW

File: src/instruction.cpp [hunk 1]
```diff
@@ -74,14 +74,18 @@ bool operator==(const instruction& i, instruction_ref ref)
     return std::addressof(i) == std::addressof(*ref);
 }
 
-bool instruction::valid(instruction_ref start, bool check_order) const
+static void debug_name(std::ostream& os, const instruction& ins);
```
Comment (kahmed10) [inline src/instruction.cpp:77]: Why does this need to be defined here?

Comment (kahmed10) [inline src/instruction.cpp:77]: Why does this need to be defined here?

Comment (CharlieL7) [reply src/instruction.cpp:77]: Leftover from debugging, removed.

**Comment by CharlieL7** [comment]
Tried some tests, the hanging error is actually a little more nuanced. Making a new PR to explain it better.

**Pull request closed without merging**